src/Arcus.Messaging.Health/Tcp/TcpHealthListener.cs

```diff
@@ -137,6 +144,29 @@ private async Task AcceptConnectionAsync()
             }
         }
 
+        private byte[] SerializeHealthReport(HealthReport healthReport)
+        {
+            IHealthReportSerializer reportSerializer = _tcpListenerOptions.HealthReportSerializer;
+            if (reportSerializer is null)
```

Comment:
THinking out loud - If with 2 implementations, split it in two methods or not?

Reply:
Normally, yes. But I think in this case the implementation is too small to really benefit from it?

Reply:
👌